expression/integration_test.go

```diff
@@ -8257,3 +8257,11 @@ func (s *testIntegrationSuite) TestIssue12205(c *C) {
 	tk.MustQuery("show warnings;").Check(
 		testkit.Rows("Warning 1292 Truncated incorrect time value: '18446744072635875000'"))
 }
+
+func (s *testIntegrationSuite) TestIssue11333(c *C) {
+	defer s.cleanEnv(c)
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustQuery(`select 0.00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000;`).Check(testkit.Rows("0.000000000000000000000000000000000000000000000000000000000000000000000000"))
```

Comment:
What about the behavior in `Insert` statement?

Reply:
The value will truncated decided by  `D` of `decimal(M,D)`.
```
mysql> create table t(col1 decimal);
Query OK, 0 rows affected (0.01 sec)

mysql> insert into t values(0.00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000);
Query OK, 1 row affected (0.01 sec)

mysql> select * from t;
+------+
| col1 |
+------+
|    0 |
+------+
1 row in set (0.01 sec)

mysql> show create table t;
+-------+--------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                       |
+-------+--------------------------------------------------------------------------------------------------------------------+
| t     | CREATE TABLE `t` (
  `col1` decimal(10,0) DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
+-------+--------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

mysql> create table t1(col1 decimal(65,30));
Query OK, 0 rows affected (0.01 sec)

mysql> insert into t1 values(0.00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000);
Query OK, 1 row affected (0.00 sec)

mysql> select * from t1;
+----------------------------------+
| col1                             |
+----------------------------------+
| 0.000000000000000000000000000000 |
+----------------------------------+
1 row in set (0.00 sec)
```

The behavior is same with `MySQL-8.0.19`. Before this pr, tidb will throw an error.
```
mysql> create table t(col1 decimal);
Query OK, 0 rows affected (0.14 sec)

mysql> insert into t values(0.00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000);
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 112 near "0.00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000)"decimal literal: [types:1265]Data truncated for column '%s' at row %d
```

Reply:
It'll be nice to add these cases.

Reply:
OK, I will add these cases later.